astropy/stats/circstats.py

```diff
@@ -0,0 +1,555 @@
+        Array of circular (directional) data.
+    deg : boolean, optional
+        If ``data`` is expressed in degrees, this should be ``True``.
+        If ``deg`` is ``False``, the data is assumed to be measured in radians.
+    w_alpha : numpy.ndarray, optional
+        In case of grouped data, the i-th element of ``w_alpha`` represents a
+        weighting factor for each group such that sum(w,axis) equals the number
+        of observations. See [1], remark 1.4, page 22, for detailed
+        explanation.
+    w_beta : numpy.ndarray, optional
+        See description of ``w_alpha``.
+    ax_alpha : int, optional
+        Axis along which circular correlation coefficients are computed.
+        The default is the compute the circular correlation coefficient of the
+        flattened array.
+    ax_ beta : int, optional
+        See description of ``ax_alpha``.
+
+    Returns
+    -------
+    rho : float
+        Circular correlation coefficient.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from astropy.stats import circcorrcoef
+    >>> alpha_deg = np.array([356, 97, 211, 232, 343, 292, 157, 302, 335, 302,
+    ...                       324, 85, 324, 340, 157, 238, 254, 146, 232, 122,
+    ...                       329])
+    >>> beta_deg = np.array([119, 162, 221, 259, 270, 29, 97, 292, 40, 313, 94,
+    ...                      45, 47, 108, 221, 270, 119, 248, 270, 45, 23])
+    >>> alpha_rad = np.deg2rad(alpha_deg)
+    >>> beta_rad = np.deg2rad(beta_deg)
+    >>> circcorrcoef(alpha_rad, beta_rad) # doctest: +FLOAT_CMP
+    <Quantity 0.2704648826748831>
+    >>> circcorrcoef(alpha_deg, beta_deg, deg=True) # doctest: +FLOAT_CMP
+    <Quantity 0.2704648826748831>
+    >>> from astropy import units as u
+    >>> circcorrcoef(alpha_rad*u.rad, beta_rad*u.rad) # doctest: +FLOAT_CMP
+    <Quantity 0.2704648826748831>
+    >>> circcorrcoef(alpha_deg*u.deg, beta_deg*u.deg,
+    ...              deg=True) # doctest: +FLOAT_CMP
+    <Quantity 0.2704648826748831>
+
+    References
+    ----------
+    .. [1] S. R. Jammalamadaka, A. SenGupta. "Topics in Circular Statistics".
+       Series on Multivariate Analysis, Vol. 5, 2001.
+    .. [2] C. Agostinelli, U. Lund. "Circular Statistics from 'Topics in
+       Circular Statistics (2001)'". 2015.
+       <https://cran.r-project.org/web/packages/CircStats/CircStats.pdf>
+    """
+    if(np.size(alpha, axis=ax_alpha) != np.size(beta, axis=ax_beta)):
+        raise ValueError("alpha and beta must be arrays of the same size")
+
+    if deg:
+        alpha = Quantity(alpha, u.deg)
+        beta = Quantity(beta, u.deg)
+    else:
+        alpha = Quantity(alpha, u.rad)
+        beta = Quantity(beta, u.rad)
+
+    mu_a = circmean(alpha, deg, w_alpha, ax_alpha)
+    mu_b = circmean(beta, deg, w_beta, ax_beta)
+
+    sin_a = np.sin(alpha - mu_a)
+    sin_b = np.sin(beta - mu_b)
+    rho = np.sum(sin_a*sin_b)/np.sqrt(np.sum(sin_a*sin_a)*np.sum(sin_b*sin_b))
+
+    return rho
```

Comment:
I agree with @crawfordsm's general comment that the `deg` flag should be removed. It will make the code much cleaner; and fine *not* to convert to `Quantity` at all, so that `ndarray` just gets treated as radians. In that case, I also think it is fine for the output here to just be of the type of the input, i.e., dimensionless `Quantity` if `Quantity` input was used.